Store/Get-LargeMailboxFolderStatistics.ps1

```diff
@@ -0,0 +1,68 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+<#
+.SYNOPSIS
+    Retrieves folder statistics for mailboxes with large numbers of folders.
+
+.DESCRIPTION
+    This script retrieves folder statistics for large mailboxes or a specified user identity.
+    It can target either primary or archive mailboxes and processes the data in batches.
+
+.PARAMETER Identity
+    The identity of the user whose mailbox folder statistics are to be retrieved.
+
+.PARAMETER MailboxType
+    Specifies the type of mailbox to target. Valid values are "Primary" and "Archive".
+    Default is "Archive".
+
+.PARAMETER BatchSize
+    Specifies the number of items to process in each batch. Default is 5000.
+
+.PARAMETER Properties
+    Specifies the properties to include in the output. Default is "Name, FolderPath, ItemsInFolder, FolderSize, FolderAndSubfolderSize".
+
+.EXAMPLE
+    $folderStats = .\Get-LargeMailboxFolderStatistics.ps1 -Identity fred@contoso.com
+    $folderStats = .\Get-LargeMailboxFolderStatistics.ps1 -Identity fred@contoso.com -MailboxType Primary
+    $folderStats = .\Get-LargeMailboxFolderStatistics.ps1 -Identity fred@contoso.com -MailboxType Archive -BatchSize 5000 -Properties "Name, FolderPath"
+#>
+param(
+    [Parameter(Mandatory = $true, Position = 0)]
+    $Identity,
+    [Parameter(Mandatory = $false, Position = 1)]
+    [ValidateSet("Primary", "Archive")]
+    $MailboxType = "Archive",
+    [Parameter(Mandatory = $false, Position = 2)]
+    $BatchSize = 5000,
+    [Parameter(Mandatory = $false, Position = 3)]
+    $Properties = "Name, FolderPath, ItemsInFolder, FolderSize, FolderAndSubfolderSize"
+)
+
+Process {
+    $allContentFolders = $null
+    $propertyArray = $Properties -split ',' | ForEach-Object { $_.Trim() }
```

Comment:
Would be best to have the parameter handle this already and accept it as a string array instead of doing this.

Reply:
Updated